src/vorta/utils.py

```diff
@@ -191,35 +194,22 @@ def get_private_keys():
             key_file = os.path.join(ssh_folder, key)
             if not os.path.isfile(key_file):
                 continue
-            for key_format in key_formats:
-                try:
-                    parsed_key = key_format.from_private_key_file(key_file)
-                    key_details = {
-                        'filename': key,
-                        'format': parsed_key.get_name(),
-                        'bits': parsed_key.get_bits(),
-                        'fingerprint': parsed_key.get_fingerprint().hex(),
-                    }
-                    available_private_keys.append(key_details)
-                except (
-                    SSHException,
-                    UnicodeDecodeError,
-                    IsADirectoryError,
-                    IndexError,
-                    ValueError,
-                    PermissionError,
-                    NotImplementedError,
-                ):
-                    logger.debug(
-                        f'Expected error parsing file in .ssh: {key} (You can safely ignore this)', exc_info=True
-                    )
+            # ignore config, known_hosts*, *.pub, etc.
+            if key.endswith('.pub') or key.startswith('known_hosts') or key == 'config':
+                continue
+            try:
+                if is_ssh_file(key_file):
+                    available_private_keys.append(key)
+            except (PermissionError, UnicodeDecodeError):
+                # Handling PermissionError separately from OSError because it can be safely ignored
+                # (If the user doesn't have permission to read the file, it's not an unexpected error)
+                # If UnicodeDecodeError is raised, it's because the file is not a valid SSH key
+                logger.debug(f'Expected error parsing file in .ssh: {key} (You can safely ignore this)', exc_info=True)
+                continue
+            except OSError as e:
+                if e.errno == errno.ENXIO:
+                    # when key_file is a (ControlPath) socket
                     continue
-                except OSError as e:
-                    if e.errno == errno.ENXIO:
-                        # when key_file is a (ControlPath) socket
-                        continue
-                    else:
-                        raise
```

Comment:
You have an "if", but you dropped the "else: raise".

The "continue" in the if-block basically means "do nothing" and as you have removed the else-block, it is also doing nothing there.

Looks suspicious.

Reply:
So yeah, I guess I should raise the error in the end.